File: client-scalaz/src/main/scala/org/openapitools/client/api/TutorialResponse.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import TutorialResponse._

case class TutorialResponse (
  valid: Option[Boolean],
message: Option[String],
version: Option[Double],
serializeNulls: Option[Boolean],
startTimeLog: Option[Long],
errorCode: Option[String],
request: Option[List[NameStringValueResponse]],
id: Option[Long],
name: Option[String],
description: Option[String],
alignment: Option[String],
image: Option[AssetShortResponse],
orderIndex: Option[Long],
objectType: Option[String],
returning: Option[String])

object TutorialResponse {
  import DateTimeCodecs._

  implicit val TutorialResponseCodecJson: CodecJson[TutorialResponse] = CodecJson.derive[TutorialResponse]
  implicit val TutorialResponseDecoder: EntityDecoder[TutorialResponse] = jsonOf[TutorialResponse]
  implicit val TutorialResponseEncoder: EntityEncoder[TutorialResponse] = jsonEncoderOf[TutorialResponse]
}

File: client-clojure/src/sirqul_io_t_platform/specs/tutorial_response.clj
(ns sirqul-io-t-platform.specs.tutorial-response
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [sirqul-io-t-platform.specs.name-string-value-response :refer :all]
            [sirqul-io-t-platform.specs.asset-short-response :refer :all]
            )
  (:import (java.io File)))


(def tutorial-response-data
  {
   (ds/opt :valid) boolean?
   (ds/opt :message) string?
   (ds/opt :version) float?
   (ds/opt :serializeNulls) boolean?
   (ds/opt :startTimeLog) int?
   (ds/opt :errorCode) string?
   (ds/opt :request) (s/coll-of name-string-value-response-spec)
   (ds/opt :id) int?
   (ds/opt :name) string?
   (ds/opt :description) string?
   (ds/opt :alignment) string?
   (ds/opt :image) asset-short-response-spec
   (ds/opt :orderIndex) int?
   (ds/opt :objectType) string?
   (ds/opt :returning) string?
   })

(def tutorial-response-spec
  (ds/spec
    {:name ::tutorial-response
     :spec tutorial-response-data}))
